src/stash/src/transaction.rs

```diff
@@ -31,6 +31,17 @@ use crate::{
     StashError, Timestamp,
 };
 
+// The limit AFTER which to split an update batch (that is, we will ship an update that
+// exceeds this number, but then start another batch).
+//
+// Cockroach's default limit for sql.conn.max_read_buffer_message_size is 16MB
+// (https://github.com/cockroachdb/cockroach/blob/7e4e0b195cd61da6cd7a719a5b9aa2e84f68d475/pkg/sql/pgwire/pgwirebase/encoding.go#L50).
+// Use a number well under that but still big ish that most things won't ever need to
+// batch. Because we are only estimating the value size and ignoring various other
+// things that contribute to the total pgwire message size, having a 14MB headspace
+// seems safe here.
+pub const INSERT_BATCH_SPLIT_SIZE: usize = 2_000_000;
```

Comment:
Tiniest nit, based on this code link, Cockroach's limit is actually 16 **MiB**. It doesn't really matter but maybe we could make our limit 2MiB, so everything is the same factor? e.g. `2 * 1024 * 1024`?